indico/MaKaC/authentication/baseAuthentication.py

```diff
@@ -287,25 +289,30 @@ def retrieveAvatar(self, rh):
             if personId == '-1':
                 personId = None
             ah = AvatarHolder()
-            av = ah.match({"email": email}, exact=1, onlyActivated=False, searchInAuthenticators=False)
-            if av:
-                av = av[0]
-                # don't allow disabled accounts
-                if av.isDisabled():
-                    return None
-                elif not av.isActivated():
-                    av.activateAccount()
-
-                av.clearAuthenticatorPersonalData()
-                av.setAuthenticatorPersonalData('phone', phone)
-                av.setAuthenticatorPersonalData('fax', fax)
-                av.setAuthenticatorPersonalData('surName', lastname)
-                av.setAuthenticatorPersonalData('firstName', firstname)
-                av.setAuthenticatorPersonalData('affiliation', institute)
-                if personId != None and personId != av.getPersonId():
-                    av.setPersonId(personId)
-            else:
-                avDict = {"email": email,
+            for mail in re.split(";|,", email):
+                av = ah.match({"email": mail}, exact=1, onlyActivated=False, searchInAuthenticators=False)
+                if av:
+                    av = av[0]
+                    # don't allow disabled accounts
+                    if av.isDisabled():
+                        return None
+                    elif not av.isActivated():
+                        av.activateAccount()
+
+                    av.clearAuthenticatorPersonalData()
+                    av.setAuthenticatorPersonalData('phone', phone)
+                    av.setAuthenticatorPersonalData('fax', fax)
+                    av.setAuthenticatorPersonalData('surName', lastname)
+                    av.setAuthenticatorPersonalData('firstName', firstname)
+                    av.setAuthenticatorPersonalData('affiliation', institute)
+                    if personId != None and personId != av.getPersonId():
+                        av.setPersonId(personId)
+
+                    self._postLogin(login, av, True)
+                    return av
+
+            if not av:
```

Comment:
Can't we use `for..else` here? And break the loop when an avatar is found, instead of calling `_postLogin`.
We could then de-dent L327 and L328 as before. Right?